File: PTT-GC-API/Models/Initiative/ImpactTracking.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PTT_GC_API.Models.Initiative
{
    public class ImpactTracking
    {
        public int Id { get; set; }
        [StringLength(200)]
        public string FinancialImpactArea { get; set; }
        public bool HaveShareBenefit { get; set; }
        [Column(TypeName = "decimal(18,3)")]
        public decimal? IL4RunRateRecurring { get; set; }
        [Column(TypeName = "decimal(18,3)")]
        public decimal? IL5RunRateRecurring { get; set; }
        [Column(TypeName = "decimal(18,3)")]
        public decimal? IL4RunRateOnetime { get; set; }
        [Column(TypeName = "decimal(18,3)")]
        public decimal? IL5RunRateOnetime { get; set; }
        [Column(TypeName = "decimal(18,3)")]
        public decimal? IL5FixedFxRecurring { get; set; }
        [Column(TypeName = "decimal(18,3)")]
        public decimal? IL5FloatFxRecurring { get; set; }
        [Column(TypeName = "decimal(18,3)")]
        public decimal? IL5FixedFxOnetime { get; set; }
        [Column(TypeName = "decimal(18,3)")]
        public decimal? IL5FloatFxOnetime { get; set; }
        [Column(TypeName = "decimal(18,3)")]
        public decimal? TotalRecurring { get; set; }
        [Column(TypeName = "decimal(18,3)")]
        public decimal? TotalOnetime { get; set; }
        public DateTime? FirstRunRateMonth { get; set; }
        public bool AutoCalculate { get; set; }
        public bool IndirectBenefit { get; set; }
        [Column(TypeName = "nvarchar(MAX)")]
        public string Explanation { get; set; }
        public bool ImpiemantCost  { get; set; }
        [Column(TypeName = "decimal(18,3)")]
        public decimal? TotalCostOPEX { get; set; }
        [Column(TypeName = "nvarchar(MAX)")]
        public string ToComment { get; set; }
        [StringLength(500)]
        public string Remark1 { get; set; }
        [StringLength(500)]
        public string Remark2 { get; set; }
        [StringLength(500)]
        public string Remark3 { get; set; }
        [StringLength(500)]
        public string Remark4 { get; set; }
        [StringLength(500)]
        public string Remark5 { get; set; }
        public int InitiativeId { get; set; }
        [StringLength(50)]
        public string InitiativeCode { get; set; }
        [StringLength(10)]
        public string SIL4Achievement { get; set; }
        [StringLength(10)]
        public string SIL5Achievement { get; set; }
        [StringLength(500)]
        public string ContactIO { get; set; }
        public DateTime? LastApprovedIL4Date { get; set; }
        public DateTime? LastApprovedIL5Date { get; set; }
        [StringLength(500)]
        public string ContactIOBy { get; set; }
        public DateTime? FirstApprovedIL4Date { get; set; }
        public DateTime? LastSubmittedSIL4Date { get; set; }
        public DateTime? LastSubmittedSIL5Date { get; set; }
    }
}

File: PTT-GC-SPA/src/app/core/models/impacttracking.ts
export interface ImpactTracking {
    id: number;
    financialImpactArea: string;
    haveShareBenefit: boolean;
    iL4RunRateRecurring: number;
    iL5RunRateRecurring: number;
    iL4RunRateOnetime: number;
    iL5RunRateOnetime: number;
    iL5FixedFxRecurring: number;
    iL5FloatFxRecurring: number;
    iL5FixedFxOnetime: number;
    iL5FloatFxOnetime: number;
    totalRecurring: number;
    totalOnetime: number;
    firstRunRateMonth: Date;
    autoCalculate: boolean;
    indirectBenefit: boolean;
    explanation: string;
    impiemantCost: boolean;
    totalCostOPEX: number;
    toComment: string;
    remark1: string;
    remark2: string;
    remark3: string;
    remark4: string;
    remark5: string;
    initiativeId: number;
    initiativeCode: string;
    siL4Achievement: string;
    siL5Achievement: string;
    contactIO: string;
    lastApprovedIL4Date: Date;
    lastApprovedIL5Date: Date;
    contactIOBy: string;
    firstApprovedIL4Date: Date;
    lastSubmittedSIL4Date: Date;
    lastSubmittedSIL5Date: Date;
}
export interface FirstRunRateTable {
    id: number;
    impactTypeOfBenefitTable: string;
    typeOfBenefit: string;
    versionPrice: VersionPrice;
    runRate: RunRate;
    monthRows1: MonthRow;
    monthRows2: MonthRow;
    monthRows3: MonthRow;
}
export interface VersionPrice {
    row1: string;
    row2: string;
    row3: string;
}
export interface RunRate {
    row1: number;
    row2: number;
    row3: number;
}
export interface MonthRow {
    month1: number;
    month2: number;
    month3: number;
    month4: number;
    month5: number;
    month6: number;
    month7: number;
    month8: number;
    month9: number;
    month10: number;
    month11: number;
    month12: number;
    month13: number;
    month14: number;
    month15: number;
    month16: number;
    month17: number;
    month18: number;
    month19: number;
    month20: number;
    month21: number;
    month22: number;
    month23: number;
    month24: number;
    month25: number;
    month26: number;
    month27: number;
    month28: number;
    month29: number;
    month30: number;
    month31: number;
    month32: number;
    month33: number;
    month34: number;
    month35: number;
    month36: number;
}
export interface MonthRowDateTempData {
    month: Date;
    monthData: number;
}
export interface runRateTempData {
    runRateMonthTempMonth1: MonthRowDateTempData[];
    runRateMonthTempMonth2: MonthRowDateTempData[];
    runRateMonthTempMonth3: MonthRowDateTempData[];
}